mobile-app/lib/ui/views/learn/block/templates/grid/grid_view.dart

```diff
@@ -0,0 +1,153 @@
+
+    return ViewModelBuilder.reactive(
+      viewModelBuilder: () => BlockGridViewModel(),
+      builder: (context, childModel, child) => Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Container(
+            padding: const EdgeInsets.all(8),
+            // For some dumb reason the progress indicator does not
+            // get a specified width from the column.
+            width: MediaQuery.of(context).size.width * 0.9,
+            child: LinearProgressIndicator(
+              minHeight: 10,
+              value: progress,
+              valueColor: const AlwaysStoppedAnimation<Color>(
+                Color.fromRGBO(0x99, 0xc9, 0xff, 1),
+              ),
+              backgroundColor: const Color.fromRGBO(0x2a, 0x2a, 0x40, 1),
+              borderRadius: BorderRadius.circular(10),
+            ),
+          ),
+          Row(
+            children: [
+              Padding(
+                padding: const EdgeInsets.all(8),
+                child: TextButton(
+                  onPressed: () {
+                    model.setIsOpen = !model.isOpen;
+                  },
+                  style: TextButton.styleFrom(
+                    backgroundColor: const Color.fromRGBO(0x1b, 0x1b, 0x32, 1),
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(5),
+                      side: const BorderSide(
+                        color: Color.fromRGBO(0x3b, 0x3b, 0x4f, 1),
+                      ),
+                    ),
+                  ),
+                  child: Text(
+                    model.isOpen ? 'Hide Steps' : 'Show Steps',
+                  ),
+                ),
+              ),
+            ],
+          ),
+          if (model.isOpen)
+            Row(
+              children: [
+                Container(
+                  padding: const EdgeInsets.all(8),
+                  height: 195,
+                  width: MediaQuery.of(context).size.width - 34,
+                  child: GridView.builder(
+                    gridDelegate:
+                        const SliverGridDelegateWithFixedCrossAxisCount(
+                      crossAxisCount: 6,
+                      mainAxisExtent: 60,
+                      mainAxisSpacing: 3,
+                      crossAxisSpacing: 3,
+                    ),
+                    itemCount: block.challenges.length,
+                    itemBuilder: (context, step) {
+                      return ChallengeTile(
+                        block: block,
+                        model: model,
+                        step: step + 1,
+                        challengeId: block.challengeTiles[step].id,
+                        isDownloaded: false,
+                      );
+                    },
+                  ),
+                ),
+              ],
+            ),
+        ],
+      ),
+    );
+  }
+}
+
+class ChallengeTile extends StatelessWidget {
```

Comment:
Isn't this the same as the `ChallengeTile` in dialogue view template? If so we could create a common widget for it.